### ISSUE TITLE
Add support for 32-bit

### ISSUE THREAD

**Issue by Esption** [godot-rust/gdext#347]
Trying to do `cargo build --target i686-pc-windows-msvc` fails to build.  It is possible to go in and get it to manually compile by editing `godt-ffi/src/opaque.rs` to have an `align(4)` instead of `8`, and the `OpaqueObject` in `godot-ffi/src/gen/central.rs` from `8uszie` to `4usize` (you need to wait until *after* you see the fail build for that, because of codegen). Likely there are other things in that need to be updated as well, but at a minimum that will successfully cause it compile.  I wasn't able to figure out where that codegen is actually happening.  After editing those two things, the creeps example will start and run, but appears to crash on the `anim_names.to_vec()` in the `rust/src/mob.rs`

**Comment by lilizoey** [comment]
i think we need to change this:
```rs
    // For float/double inference, see:
    // * https://github.com/godotengine/godot-proposals/issues/892
    // * https://github.com/godotengine/godot-cpp/pull/728
    #[cfg(feature = "double-precision")]
    let build_config = "double_64"; // TODO infer this
    #[cfg(not(feature = "double-precision"))]
    let build_config = "float_64"; // TODO infer this
    ```
  in [here](https://github.com/godot-rust/gdext/blob/fd1d50fcf826ad228f96c5a8e5bf8461e1c9d990/godot-codegen/src/api_parser.rs#L244) to the respective `x_32` build configs for this.

**Comment by Esption** [comment]
So, I'm pretty sure the codegen for the Opaques originates from `godot4-prebuilt::load_gdextension_json()` called in `godot-codegen/src/lib.rs` 
I guess just need to give that a different target?

**Comment by lilizoey** [comment]
yea probably, we can likely use https://doc.rust-lang.org/reference/conditional-compilation.html#target_pointer_width this cfg flag to decide between which representation to use

**Comment by Esption** [comment]
Not sure if should mention here or over there, but
https://github.com/godot-rust/godot4-prebuilt
seems to explictly generate for 64-bit targets.  It *seems* fine for the most part with one exception.  The built-in tests will fail due to the widths assuming 64-bit.  If we added other targets, it would be nice to get win32/linux32/wasm added in there. Example command is to just run `cargo test --no-default-features --target i686-pc-windows-msvc`

Best I can tell, that's the only reason the itest suite fails as a whole against 32-bit targets, but hard to tell atm.

**Comment by Bromeon** [comment]
So this would need:
- a `godot4-prebuilt` configuration which generates artifacts for 32-bit
- a CI job in gdext that runs unit tests for it
Anything else?

Probably it's good enough to combine this with WASM for now...

**Comment by Esption** [comment]
AFAIK yes that should be the final thing to tick off 32-bit support.

**Comment by Ughuuu** [comment]
> [here](https://github.com/godot-rust/gdext/blob/fd1d50fcf826ad228f96c5a8e5bf8461e1c9d990/godot-codegen/src/api_parser.rs#L244)

So for web would it be float_32? Is this issue also related to if someone is building locally with `api-custom`? Because I have error with that also.

**Comment by Bromeon** [comment]
Yes. If someone could test how to invoke bindgen so that `api-custom` works with Wasm, and report what's needed (or open a PR), that would be very helpful 🙏

Is it just the build target, or bindgen itself needs different setup? 

Also, in the past this didn't seem to be necessary 🤔
